### ISSUE TITLE
DynamoDB looping ScanCommand crashes when ExclusiveStartKey lands on key entry with multi-byte char

### ISSUE THREAD

**Issue by codestart** [aws/aws-sdk-js-v3#1856]
**Describe the bug**
I am doing a full table scan looking through the items in a table x number at a time using `Limit` and `ExclusiveStartKey`.
When the Primary Key used by `ExclusiveStartKey` has a multi-byte value, the application crashes with the error below.

I have run the full table scan with a different `Limit` number and all data (including multi-byte chars) are retrieved.
This issue only occurs if the `Limit` number causes the `ExclusiveStartKey` to land on a key value with a multi-byte character.

I have the following code:
```
const params = {
    ExclusiveStartKey: undefined,
    Limit: 2,
    TableName: TABLE_TO_SCAN,
};

do {
  const data = await dbclient.send(new ScanCommand(params));
  params.ExclusiveStartKey = data.LastEvaluatedKey;
} while (params.ExclusiveStartKey !== undefined);
```

**Data needed**
One of my database entries contains a key with `á` it it.

I am using `"@aws-js-sdk-v3-prerelease/client-dynamodb-v2-node": "0.1.0-preview.2"`

Running Typescript on NodeJs/ts-node: v9.1.1 (on Windows 10)

**Very similar to this closed issue**
https://github.com/aws/aws-sdk-js-v3/issues/177

**Resulting Error Message**
```
Error SerializationException: 
    at JsonRpcParser.exports.jsonErrorUnmarshaller [as parseServiceException] (<app_location>\<app_name>\node_modules\@aws-js-sdk-v3-prerelease\json-error-unmarshaller\src\index.ts:65:33)
    at JsonRpcParser.<anonymous> (<app_location>\<app_name>\node_modules\@aws-js-sdk-v3-prerelease\protocol-json-rpc\src\JsonRpcParser.ts:33:24)
    at step (<app_location>\<app_name>\node_modules\tslib\tslib.js:141:27)
    at Object.next (<app_location>\<app_name>\node_modules\tslib\tslib.js:122:57)
    at fulfilled (<app_location>\<app_name>\node_modules\tslib\tslib.js:112:62)
    at processTicksAndRejections (internal/process/task_queues.js:97:5) {
  name: 'SerializationException',
  message: '',
  details: {},
  '$metadata': {
    httpHeaders: { - Futher detailed headers in here - }
```

**Comment by codestart** [comment]
HTH:
https://github.com/aws/aws-sdk-js-v3/blob/master/clients/client-dynamodb/models/models_0.ts#L8470

**Comment by dradetsky** [comment]
@codestart so I don't actually work on the library myself, but 0.1.0-preview.2 is like 2 years old now, and in the issue you link they said there's been a fix, so if I were you I'd upgrade to v3.1.0 if possible.

I did notice that the fix for this issue should have been included in the 0.1.0-preview.2 release. Possibly you didn't upgrade the affected library to 0.1.0-preview.2. In any case, there's no way AWS is going to provide support for this version, so you should just upgrade asap.

BTW, fun fact: saying

> Typescript on NodeJs/ts-node: v9.1.1

isn't actually very helpful b/c ts-node is just a tiny wrapper tool to allow nodejs to execute typescript. You should probably say what version of nodejs you're running in the future.

**Comment by codestart** [comment]
@dradetsky Thanks for this, I was looking in npmjs.com but not changing the package name enough from what Google had given me! Finally found the version I need under: "@aws-sdk/client-dynamodb": "3.1.0". Tests passing well now.
